deploy/AllocatorInit.sol

```diff
@@ -230,92 +218,112 @@ library AllocatorInit {
 
         // Allow vault and funnels to pull funds from the buffer
         BufferLike(ilkInstance.buffer).approve(VaultLike(ilkInstance.vault).nst(), ilkInstance.vault, type(uint256).max);
+
+        // Set the allocator proxy as the ilk admin instead of the Pause Proxy
+        RolesLike(sharedInstance.roles).setIlkAdmin(ilk, cfg.allocatorProxy);
+
+        // Move ownership of the ilk contracts to the allocator proxy
+        ScriptTools.switchOwner(ilkInstance.vault,  ilkInstance.owner, cfg.allocatorProxy);
+        ScriptTools.switchOwner(ilkInstance.buffer, ilkInstance.owner, cfg.allocatorProxy);
+
+        // Add allocator-specific contracts to changelog
+        string memory ilkString = ScriptTools.ilkToChainlogFormat(ilk);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(ilkString, "_VAULT"))),  ilkInstance.vault);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(ilkString, "_BUFFER"))), ilkInstance.buffer);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked("PIP_", ilkString))), sharedInstance.oracle);
+
+        // Add to ilk registry
+        IlkRegistryLike(cfg.ilkRegistry).put({
+            _ilk    : ilk,
+            _join   : address(0),
+            _gem    : address(0),
+            _dec    : 0,
+            _class  : 5, // RWAs are class 3, D3Ms and Teleport are class 4
+            _pip    : sharedInstance.oracle,
+            _xlip   : address(0),
+            _name   : bytes32ToStr(ilk),
+            _symbol : bytes32ToStr(ilk)
+        });
+    }
+
+    function initIlkFunnel(
```

Comment:
Probably worth to add a comment to reiterate that instead of how `initIlk` is executed (via the `PauseProxy`), `initIlkFunnel` is designed to be run through the `AllocatorProxy` as in that instance the base `ilk` contracts `vault` and `buffer` will be already owned by it.

Reply:
Added